server/internal/core/application/utils.go

```diff
@@ -108,12 +109,22 @@ func (m *txRequestsQueue) push(
 		}
 	}
 
+	for _, vtxo := range recoveredVtxos {
+		for _, request := range m.requests {
+			for _, pVtxo := range request.recoveredVtxos {
+				if vtxo == pVtxo {
```

Comment:
Check only the outpoint

```suggestion
				if vtxo.VtxoKey == pVtxo.VtxoKey {
```